src/sciline/pipeline.py

```diff
@@ -750,3 +776,124 @@ def bind_and_call(
         if not return_tuple:
             return results[0]
         return results
+
+    def _repr_html_(self) -> str:
+        providers_without_parameters = (
+            (p, tuple(), v) for p, v in self._providers.items()
+        )  # type: ignore[var-annotated]
+        providers_with_parameters = (
+            (p, c, v)
+            for p in self._subproviders
+            for c, v in self._subproviders[p].items()
+        )
+
+        providers = groupby(
+            lambda p: p['kind'],
+            (
+                {
+                    'origin': p,
+                    'args': c,
+                    'value': v() if (k := _kind_of_provider(v)) != 'function' else v,
+                    'kind': k,
+                }
+                for p, c, v in chain(
+                    providers_with_parameters, providers_without_parameters
+                )
+            ),
+        )
+
+        param_tables_by_name = groupby(
+            lambda p: p['origin'].label[0].tp, providers['table']
+        )
+        param_tables = []
+        for table_name, table_cells in param_tables_by_name.items():
+            columns = groupby(lambda p: p['origin'].tp, table_cells)
+            index = [p['origin'].label[0].index for p in next(iter(columns.values()))]
+
+            param_tables.append(
+                ParamTable(
+                    table_name,
+                    {k: [v['value'] for v in g] for k, g in columns.items()},
+                    index=index,
+                )
+            )
+
+        def pretty_requirement_name(r: Key | TypeVar) -> str:
+            return '*' if isinstance(r, TypeVar) else f'{_qualname(r)}'
+
+        def pretty_provider_name(p: dict[str, Any]) -> str:
+            name = f'{_qualname(p["origin"])}'
+            if p['args']:
+                name += f'[{",".join(map(pretty_requirement_name, p["args"]))}]'
+            return name
+
+        def provider_source(p: dict[str, Any]) -> str:
+            if p['kind'] != 'function':
+                return ''
+            module = inspect.getmodule(p['value']).__name__
+            return f'''
+            <details>
+              <summary>{p['value'].__name__}</summary>
+                {module}.{p['value'].__name__}
+            </details>
+            '''
+
+        def provider_value(p: dict[str, Any]) -> str:
+            if p['kind'] != 'parameter':
+                return ''
+            if hasattr(p['value'], '_repr_html_'):
+                return f'{p["value"]._repr_html_()}'
+            return f'{p["value"]}'
+
+        return f'''
+        <div style="display: flex; column-gap: 1em; align-items: flex-start;">
```

Comment:
Should we maybe use template instead of f-string here...?
Then it'll be much easier to update/fix them later.
Llike for the data group html wrapper: https://github.com/scipp/scipp/blob/main/src/scipp/visualization/formatting_datagroup_html.py

Or maybe later if we want to add more columns...!

Reply:
I agree the code needs a clean up. I'll collect the inline styles and remove excessive duplication and nested f-strings etc. I'm not sure that the refactor will involve templates, I'll look into it but it might be overkill here.

Reply:
Maybe you can also move this to a different file to not grow `pipeline.py` too much?